### ISSUE TITLE
Unknown error (0x80092012)

### ISSUE THREAD

**Issue by goldfingyr** [hashicorp/vagrant#13102]
PS C:\VM\Vagrant Projects\StockUbuntu> **vagrant init generic/ubuntu2204**
A `Vagrantfile` has been placed in this directory. You are now
ready to `vagrant up` your first virtual environment! Please read
the comments in the Vagrantfile as well as documentation on
`vagrantup.com` for more information on using Vagrant.
PS C:\VM\Vagrant Projects\StockUbuntu> **vagrant up**
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'generic/ubuntu2204' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
The box 'generic/ubuntu2204' could not be found or
could not be accessed in the remote catalog. If this is a private
box on HashiCorp's Vagrant Cloud, please verify you're logged in via
`vagrant login`. Also, please double-check the name. The expanded
URL and error message are shown below:

URL: ["https://vagrantcloud.com/generic/ubuntu2204"]
Error: schannel: next InitializeSecurityContext failed: Unknown error (0x80092012) - The revocation function was unable to check revocation for the certificate.
PS C:\VM\Vagrant Projects\StockUbuntu>

### Debug output

NA

### Expected behavior

Starting an Ubuntu machine

### Actual behavior

Crapped out

#### Vagrant version

PS C:\VM\Vagrant Projects\StockUbuntu> vagrant -v
**Vagrant 2.3.4**
#### Host operating system

Windows 10 Pro

#### Guest operating system

Linux Ubuntu 22.04 LTS

#### Steps to reproduce

1. vagrant init generic/ubuntu2204
2. vagrant up
3.

#### Vagrantfile

Generated by vagrant itself

**Comment by chrisroberts** [comment]
Hi there,

It appears this may be the result of a proxy in place on your network. You should be able to workaround it using the [box_download_options](https://developer.hashicorp.com/vagrant/docs/vagrantfile/machine_settings#config-vm-box_download_options) and the `--ssl-no-revoke` curl option. The configuration in the Vagrantfile would look like:

```ruby
Vagrant.configure("2") do |config|
  ...
  config.vm.box_download_options = {"ssl-no-revoke" => true}
end
```

**Comment by goldfingyr** [comment]
Sorry - no such luck. I have been using vagrant for more that 4 years and never had this issue before, and no changes to my network, which has a static IP as well for the last 12 years. So it must be a change in the vagrant system either in the application or the repository setup.
But not checking if the certificate has been revoked makes it work even if I am not really happy with introducing this new lowering of security.
So thanks for the workaround and looking forward to a fix
Cheers, Karsten

**Comment by Outrovurt** [comment]
Hi, I'm having exactly the same issue with the latest version of vagrant (2.3.4) which I just downloaded. My previous version, which I had downloaded maybe a year ago, was working just fine. However this version produces the same error as the OP has highlighted whenever I try anything like `vagrant box add ubuntu/focal64` or `vagrant up`. The workaround does work, but it doesn't explain why the issue happens in the first place.

I definitely don't have any proxy since I am just on my home network, so that isn't the issue here.

Has anyone been able to diagnose the issue and determine what the root cause is? The error does have something to do with `curl`, but since `vagrant` ships with `curl`, then it should be using the version of `curl` it ships with, correct? And why is it that only the OP, me, and a small handful of people are having this issue? Is there anything I can do at my end to diagnose it?

In terms of using the workaround, what are the downsides in terms of it being less secure, in practical terms?

Thanks

**Comment by chrisroberts** [comment]
Hi everyone,

The new release of Vagrant includes an updated build of curl which may resolve this issue. Can you please try the new version and see if the error is still encountered? 

Thanks!

**Comment by Outrovurt** [comment]
Hi,

Thanks for the update.

I forgot to mention that I am seeing this issue on Windows 10 Home Edition.

I just installed Vagrant 2.3.6 and tried the following:

```vagrant box add ubuntu/jammy64```

Unfortunately I got the following response:

```
The box 'ubuntu/jammy64' could not be found or
could not be accessed in the remote catalog. If this is a private
box on HashiCorp's Vagrant Cloud, please verify you're logged in via
`vagrant login`. Also, please double-check the name. The expanded
URL and error message are shown below:

URL: ["https://vagrantcloud.com/ubuntu/jammy64"]
Error: schannel: next InitializeSecurityContext failed: Unknown error (0x80092012) - The revocation function was unable to check revocation for the certificate.
```

If I then add the `--insecure` option, it proceeds to download just fine.

The same thing happens with the Vagrantfile, which requires the following entry for it to work:

```config.vm.box_download_insecure = true```

I don't quite know how `vagrant` is calling `curl`, however I found something peculiar on my own system. If I run `curl` on the command-line without any arguments, I get:

```
cmdlet Invoke-WebRequest at command pipeline position 1
Supply values for the following parameters:
Uri:
```

However if I run `curl.exe` (note the explicit .exe extension), I get:

`curl: try 'curl --help' for more information`

In other words, it appears as if there might be *two curls* running on my system. Calling where.exe I find that there is a curl.exe located at `C:\Windows\System32\curl.exe`, however I have no idea where the other curl is, assuming there even is another curl. Could it be a shell built-in?

Is there a possibility vagrant is calling either of these other curls and that is what is causing the problem?

**Comment by chrisroberts** [comment]
Hi there,

Would you please confirm if this issue has been resolved with the latest [Vagrant release](https://www.vagrantup.com/downloads) (version 2.3.7)?

Thanks!

**Comment by john1726** [comment]
It is working for me in 2.3.7, thank you!

**Comment by Toefer** [comment]
The issue still exists using 2.3.7:
```
> vagrant version
Installed Version: 2.3.7
Latest Version: 2.3.7

You're running an up-to-date version of Vagrant!
```
```
> vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Box 'generic/ubuntu2204' could not be found. Attempting to find and install...
    default: Box Provider: virtualbox
    default: Box Version: >= 0
There was an error while downloading the metadata for this box.
The error message is shown below:

schannel: next InitializeSecurityContext failed: Unknown error (0x80092012) - The revocation function was unable to check revocation for the certificate.
```

**Comment by Outrovurt** [comment]
Likewise, the same issue is there on 2.3.7:

```
> vagrant box update
==> default: Checking for updates to 'ubuntu/jammy64'
    default: Latest installed version: 20230606.0.0
    default: Version constraints:
    default: Provider: virtualbox
There was an error while downloading the metadata for this box.
The error message is shown below:

schannel: next InitializeSecurityContext failed: Unknown error (0x80092012) - The revocation function was unable to check revocation for the certificate.
```

And a similar error occurs when I try `vagrant up`.

**Comment by Outrovurt** [comment]
@chrisroberts can you please clarify how I can use `box_download_disable_ssl_revoke_best_effort` to fix this issue? Thanks.

**Comment by Outrovurt** [comment]
Yes, that solved this issue for me, thanks a lot.
